sphinx/builders/linkcheck.py

```diff
@@ -617,7 +625,7 @@ def _get_request_headers(
     return {}
 
 
-def contains_anchor(response: Response, anchor: str) -> bool:
+def contains_anchor(response: Response, anchor: str, *, lenient: bool = False) -> bool:
```

Comment:
I now think that the way to progress this pull request is to:

  * Revert the introduction of any notion of lenient parsing, allowing the removal of the related config setting.
  * Provide a replacement for this helper function named `check_response` or similar.  Initially the replacement function should accept a single, optional `anchor` element -- but it's worth bearing in mind the possibility to check for multiple anchors in a single response to resolve #4303.

Reply:
(note: I'm not planning to attempt this for 7.4.x - it doesn't seem high priority)